server/src/main/java/io/crate/types/DataTypes.java

```diff
@@ -353,6 +353,26 @@ private static boolean safeConversionPossible(DataType type1, DataType type2) {
         return dataType;
     }
 
+    public static DataType<?> of(String typeName, List<Integer> parameters) {
+        DataType<?> dataType = ofNameOrNull(typeName);
+        if (dataType == null) {
+            throw new IllegalArgumentException("Cannot find data type: " + typeName);
+        }
+        if (!parameters.isEmpty()) {
+            if (dataType.id() == StringType.ID) {
+                if (parameters.size() != 1) {
+                    throw new IllegalArgumentException(
+                        "The number of parameters for the text data is wrong: " + parameters.size());
+                }
+                return StringType.of(parameters.get(0));
```

Comment:
Maybe have a `of` overload for `StringType` that takes a list and move the validation there? Seems to me that the `StringType` should be responsible for knowing valid parameters, not `DataTypes`.